In [2]:
#Importing required libraries
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

In [3]:
#Importing dataset (Daily Activity)
df=pd.read_csv('Daily Activity.csv')
df.head()

Id ActivityDate  TotalSteps  TotalDistance  TrackerDistance  \
0  1503960366   04-12-2016       13162           8.50             8.50   
1  1624580081   04-12-2016        8163           5.31             5.31   
2  1644430081   04-12-2016       10694           7.77             7.77   
3  1844505072   04-12-2016        6697           4.43             4.43   
4  1927972279   04-12-2016         678           0.47             0.47   

   LoggedActivitiesDistance  VeryActiveDistance  ModeratelyActiveDistance  \
0                       0.0                1.88                      0.55   
1                       0.0                0.00                      0.00   
2                       0.0                0.14                      2.30   
3                       0.0                0.00                      0.00   
4                       0.0                0.00                      0.00   

   LightActiveDistance  SedentaryActiveDistance  VeryActiveMinutes  \
0                 6.06                      0.0                 25   
1                 5.31                      0.0                  0   
2                 5.33                      0.0                  2   
3                 4.43                      0.0                  0   
4                 0.47                      0.0                  0   

   FairlyActiveMinutes  LightlyActiveMinutes  SedentaryMinutes  Calories  \
0                   13                   328               728      1985   
1                    0                   146              1294      1432   
2                   51                   256              1131      3199   
3                    0                   339              1101      2030   
4                    0                    55               734      2220   

       Day  
0  Tuesday  
1  Tuesday  
2  Tuesday  
3  Tuesday  
4  Tuesday

In [13]:
#Calculating some statistical data
df.describe()

Id    TotalSteps  TotalDistance  TrackerDistance  \
count  9.400000e+02    940.000000     940.000000       940.000000   
mean   4.855407e+09   7637.910638       5.489702         5.475351   
std    2.424805e+09   5087.150742       3.924606         3.907276   
min    1.503960e+09      0.000000       0.000000         0.000000   
25%    2.320127e+09   3789.750000       2.620000         2.620000   
50%    4.445115e+09   7405.500000       5.245000         5.245000   
75%    6.962181e+09  10727.000000       7.712500         7.710000   
max    8.877689e+09  36019.000000      28.030000        28.030000   

       LoggedActivitiesDistance  VeryActiveDistance  ModeratelyActiveDistance  \
count                940.000000          940.000000                940.000000   
mean                   0.108128            1.502681                  0.567543   
std                    0.619725            2.658941                  0.883580   
min                    0.000000            0.000000                  0.000000   
25%                    0.000000            0.000000                  0.000000   
50%                    0.000000            0.210000                  0.240000   
75%                    0.000000            2.052500                  0.800000   
max                    4.940000           21.920000                  6.480000   

       LightActiveDistance  SedentaryActiveDistance  VeryActiveMinutes  \
count           940.000000               940.000000         940.000000   
mean              3.340819                 0.001606          21.164894   
std               2.040655                 0.007346          32.844803   
min               0.000000                 0.000000           0.000000   
25%               1.945000                 0.000000           0.000000   
50%               3.365000                 0.000000           4.000000   
75%               4.782500                 0.000000          32.000000   
max              10.710000                 0.110000         210.000000   

       FairlyActiveMinutes  LightlyActiveMinutes  SedentaryMinutes  \
count           940.000000            940.000000        940.000000   
mean             13.564894            192.812766        991.210638   
std              19.987404            109.174700        301.267437   
min               0.000000              0.000000          0.000000   
25%               0.000000            127.000000        729.750000   
50%               6.000000            199.000000       1057.500000   
75%              19.000000            264.000000       1229.500000   
max             143.000000            518.000000       1440.000000   

          Calories  
count   940.000000  
mean   2303.609574  
std     718.166862  
min       0.000000  
25%    1828.500000  
50%    2134.000000  
75%    2793.250000  
max    4900.000000

 - The **average daily step count** is **7,637**, and we can gain more insights by examining the data on an hourly basis.
 - Some records indicate a minimum of 0 total steps, which seems implausible as it suggests the participants didn't move at all. This discrepancy may be due to input errors or inconsistencies in device tracking, considering that this data was collected through surveys.

In [4]:
#Importing dataset (Hourly Activity)
df2=pd.read_csv('Hourly Activity.csv')
df2.head()

Id       Date      Time  StepTotal  TotalIntensity  \
0  1503960366  4/12/2016  00:00:00        373              20   
1  1503960366  4/12/2016  01:00:00        160               8   
2  1503960366  4/12/2016  02:00:00        151               7   
3  1503960366  4/12/2016  03:00:00          0               0   
4  1503960366  4/12/2016  04:00:00          0               0   

   AverageIntensity  Calories  
0          0.333333        81  
1          0.133333        61  
2          0.116667        59  
3          0.000000        47  
4          0.000000        48

In [5]:
#Importing dataset (Sleep)
df3=pd.read_csv('Sleep Data.csv')
df3.head()

Id    SleepDay  TotalSleepRecords  TotalMinutesAsleep  \
0  1503960366  04-12-2016                  1                 327   
1  1927972279  04-12-2016                  3                 750   
2  2026352035  04-12-2016                  1                 503   
3  3977333714  04-12-2016                  1                 274   
4  4020332650  04-12-2016                  1                 501   

   TotalTimeInBed  
0             346  
1             775  
2             546  
3             469  
4             541

In [14]:
#Calculating some statistical data
df3.describe()

Id  TotalSleepRecords  TotalMinutesAsleep  TotalTimeInBed
count  4.100000e+02         410.000000          410.000000      410.000000
mean   4.994963e+09           1.119512          419.173171      458.482927
std    2.060863e+09           0.346636          118.635918      127.455140
min    1.503960e+09           1.000000           58.000000       61.000000
25%    3.977334e+09           1.000000          361.000000      403.750000
50%    4.702922e+09           1.000000          432.500000      463.000000
75%    6.962181e+09           1.000000          490.000000      526.000000
max    8.792010e+09           3.000000          796.000000      961.000000

 - Users have a **minimum** of **one sleep session** per day and a **maximum** of **three sleep sessions**.
 - The **average duration** of sleep for users is **419 minutes**.
 - On average, users spend 458 minutes in bed.

In [6]:
#Importing dataset (Heartrate)
df4=pd.read_csv('Heartrate Data.csv')
df4.head()

Id        Date      Time  Value
0  2022484408  12-04-2016  07:21:00     97
1  2022484408  12-04-2016  07:21:05    102
2  2022484408  12-04-2016  07:21:10    105
3  2022484408  12-04-2016  07:21:20    103
4  2022484408  12-04-2016  07:21:25    101

In [16]:
#Calculating some statistical data
df4.describe()

Id          Value
count  5.194940e+05  519494.000000
mean   3.915276e+09      77.606294
std    8.012784e+08      16.922515
min    2.022484e+09      38.000000
25%    4.020333e+09      64.000000
50%    4.020333e+09      77.000000
75%    4.388162e+09      89.000000
max    5.553957e+09     203.000000

In [7]:
#Importing dataset (Weight)
df5=pd.read_csv('Weight Data.csv')
df5.head()

Id       Date      Time  WeightKg  WeightPounds   Fat    BMI  \
0  1503960366   5/2/2016  23:59:59      52.6        115.96  22.0  22.65   
1  1503960366   5/3/2016  23:59:59      52.6        115.96   NaN  22.65   
2  1927972279  4/13/2016  01:08:52     133.5        294.32   NaN  47.54   
3  2873212765  4/21/2016  23:59:59      56.7        125.00   NaN  21.45   
4  2873212765  5/12/2016  23:59:59      57.3        126.32   NaN  21.69   

   IsManualReport          LogId  
0            True  1462233599000  
1            True  1462319999000  
2           False  1460509732000  
3            True  1461283199000  
4            True  1463097599000

In [17]:
#Calculating some statistical data
df5.describe()

Id    WeightKg  WeightPounds       Fat        BMI  \
count  6.700000e+01   67.000000     67.000000   2.00000  67.000000   
mean   7.009282e+09   72.035821    158.810746  23.50000  25.185224   
std    1.950322e+09   13.923206     30.695989   2.12132   3.066962   
min    1.503960e+09   52.600000    115.960000  22.00000  21.450000   
25%    6.962181e+09   61.400000    135.360000  22.75000  23.960000   
50%    6.962181e+09   62.500000    137.790000  23.50000  24.390000   
75%    8.877689e+09   85.050000    187.500000  24.25000  25.560000   
max    8.877689e+09  133.500000    294.320000  25.00000  47.540000   

              LogId  
count  6.700000e+01  
mean   1.461772e+12  
std    7.829948e+08  
min    1.460444e+12  
25%    1.461079e+12  
50%    1.461802e+12  
75%    1.462375e+12  
max    1.463098e+12

In [8]:
#Checking null counts and dataype (df)
df.info(null_counts=True)

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 940 entries, 0 to 939
Data columns (total 16 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   Id                        940 non-null    int64  
 1   ActivityDate              940 non-null    object 
 2   TotalSteps                940 non-null    int64  
 3   TotalDistance             940 non-null    float64
 4   TrackerDistance           940 non-null    float64
 5   LoggedActivitiesDistance  940 non-null    float64
 6   VeryActiveDistance        940 non-null    float64
 7   ModeratelyActiveDistance  940 non-null    float64
 8   LightActiveDistance       940 non-null    float64
 9   SedentaryActiveDistance   940 non-null    float64
 10  VeryActiveMinutes         940 non-null    int64  
 11  FairlyActiveMinutes       940 non-null    int64  
 12  LightlyActiveMinutes      940 non-null    int64  
 13  SedentaryMinutes          940 non-null    int64  
 14  Calories  

C:\Users\Mukul\AppData\Local\Temp\ipykernel_13536\3719278618.py:2: FutureWarning: null_counts is deprecated. Use show_counts instead
  df.info(null_counts=True)


In [9]:
#Checking null counts and dataypes (df2)
df2.info(null_counts=True)

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 22099 entries, 0 to 22098
Data columns (total 7 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   Id                22099 non-null  int64  
 1   Date              22099 non-null  object 
 2   Time              22099 non-null  object 
 3   StepTotal         22099 non-null  int64  
 4   TotalIntensity    22099 non-null  int64  
 5   AverageIntensity  22099 non-null  float64
 6   Calories          22099 non-null  int64  
dtypes: float64(1), int64(4), object(2)
memory usage: 1.2+ MB


C:\Users\Mukul\AppData\Local\Temp\ipykernel_13536\2272414343.py:2: FutureWarning: null_counts is deprecated. Use show_counts instead
  df2.info(null_counts=True)


In [10]:
#Checking null counts and dataypes (df3)
df3.info(null_counts=True)

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 410 entries, 0 to 409
Data columns (total 5 columns):
 #   Column              Non-Null Count  Dtype 
---  ------              --------------  ----- 
 0   Id                  410 non-null    int64 
 1   SleepDay            410 non-null    object
 2   TotalSleepRecords   410 non-null    int64 
 3   TotalMinutesAsleep  410 non-null    int64 
 4   TotalTimeInBed      410 non-null    int64 
dtypes: int64(4), object(1)
memory usage: 16.1+ KB


C:\Users\Mukul\AppData\Local\Temp\ipykernel_13536\2404271162.py:2: FutureWarning: null_counts is deprecated. Use show_counts instead
  df3.info(null_counts=True)


In [11]:
#Checking null counts and dataypes (df4)
df4.info(null_counts=True)

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 519494 entries, 0 to 519493
Data columns (total 4 columns):
 #   Column  Non-Null Count   Dtype 
---  ------  --------------   ----- 
 0   Id      519494 non-null  int64 
 1   Date    518724 non-null  object
 2   Time    518724 non-null  object
 3   Value   519494 non-null  int64 
dtypes: int64(2), object(2)
memory usage: 15.9+ MB


C:\Users\Mukul\AppData\Local\Temp\ipykernel_13536\1433224105.py:2: FutureWarning: null_counts is deprecated. Use show_counts instead
  df4.info(null_counts=True)


In [12]:
#Checking null counts and dataypes (df5)
df5.info(null_counts=True)

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 67 entries, 0 to 66
Data columns (total 9 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   Id              67 non-null     int64  
 1   Date            67 non-null     object 
 2   Time            67 non-null     object 
 3   WeightKg        67 non-null     float64
 4   WeightPounds    67 non-null     float64
 5   Fat             2 non-null      float64
 6   BMI             67 non-null     float64
 7   IsManualReport  67 non-null     bool   
 8   LogId           67 non-null     int64  
dtypes: bool(1), float64(4), int64(2), object(2)
memory usage: 4.4+ KB


C:\Users\Mukul\AppData\Local\Temp\ipykernel_13536\2786073834.py:2: FutureWarning: null_counts is deprecated. Use show_counts instead
  df5.info(null_counts=True)


**Conclusion:**
 - Preprocessing tasks such as eliminating duplicates, correcting data types, sorting, and formatting have already been completed using **Excel**, and rest will done wherever required.
 - Due to a limited sample size, the weight data should be excluded from the analysis, even if used much interpretations can't be made out of it.
 - Sleep data has only 24 participants, while all other datasets contain 33 participants.
 - Hourly data, despite being collected separately, shares the same number of participants and data collection methods, making it suitable for aggregation into a single table.
 - Despite the mismatch in the number of participants between sleep and activity data, they can still be joined for further exploration and analysis.